ambry-store/src/main/java/com.github.ambry.store/Offset.java

```diff
@@ -0,0 +1,134 @@
+  private static final int NAME_LENGTH_SIZE = 4;
+  private static final int OFFSET_SIZE = 8;
+
+  /**
+   * Construct an Offset that refers to a position in Log.
+   * @param name the name of segment being referred to.
+   * @param offset the offset within the segment.
+   * @throws IllegalArgumentException if {@code name} is {@code null} or empty or {@code offset} < 0.
+   */
+  Offset(String name, long offset) {
+    if (name == null || name.isEmpty() || offset < 0) {
+      throw new IllegalArgumentException("Name [" + name + "] is null/empty or offset [" + offset + "] < 0");
+    }
+    this.name = name;
+    this.offset = offset;
+  }
+
+  /**
+   * Constructs an Offset from a stream.
+   * @param stream the {@link DataInputStream} that will contain the serialized form of this object.
+   * @throws IllegalArgumentException if {@code name} is {@code null} or empty or {@code offset} < 0 or if the version
+   * of the record is not recognized.
+   * @throws IOException if there are I/O problems reading from the stream.
+   */
+  static Offset fromBytes(DataInputStream stream)
+      throws IOException {
+    String name;
+    long offset;
+    int version = stream.readShort();
+    switch (version) {
+      case 0:
+        name = Utils.readIntString(stream, StandardCharsets.UTF_8);
+        offset = stream.readLong();
+        break;
+      default:
+        throw new IllegalArgumentException("Unrecognized version [" + version + "] of Offset");
+    }
+    return new Offset(name, offset);
+  }
+
+  /**
+   * @return the name of the log segment for which the offset provided by {@link #getOffset()} is valid. Guaranteed to
+   * be non-null and non-empty.
+   */
+  String getName() {
+    return name;
+  }
+
+  /**
+   * @return the offset in the log segment with name provided by {@link #getName()}. Guaranteed to be >= 0.
+   */
+  long getOffset() {
+    return offset;
+  }
+
+  /**
+   * Converts this object into its byte representation.
+   * @return the byte representation of this object.
+   */
+  byte[] toBytes() {
+    byte[] nameBytes = name.getBytes(StandardCharsets.UTF_8);
+    int size = VERSION_SIZE + NAME_LENGTH_SIZE + nameBytes.length + OFFSET_SIZE;
+    byte[] bytes = new byte[size];
+    ByteBuffer buf = ByteBuffer.wrap(bytes);
+    buf.putShort(CURRENT_VERSION);
+    buf.putInt(nameBytes.length);
+    buf.put(nameBytes);
+    buf.putLong(offset);
+    return bytes;
+  }
+
+  @Override
+  public int compareTo(Offset o) {
+    // TODO (Log Segmentation): once LogSegment is coded and can resolve the relative position of a log segment based
+    // TODO (Log Segmentation): on its name, compare names also.
+    return Long.compare(offset, o.getOffset());
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    Offset other = (Offset) o;
+    return name.equals(other.name) && offset == other.offset;
```

Comment:
since this class implements `Comparable`, I would prefer making use of `compareTo()` at this point in order to ensure that `a.equals(b) <=> a.compareTo(b) == 0`.